server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/vacation/VacationResponseSet.scala

```diff
@@ -0,0 +1,62 @@
+package org.apache.james.jmap.vacation
+
+import java.time.ZonedDateTime
+
+import org.apache.james.jmap.api.vacation.VacationPatch
+import org.apache.james.jmap.model.AccountId
+import org.apache.james.jmap.model.State.State
+import play.api.libs.json.JsObject
+
+case class VacationResponseSetRequest(accountId: AccountId,
+                               update: Map[String, VacationResponsePatchObject]) {
+  def parsePatch(): Either[IllegalArgumentException, VacationResponsePatchObject] = {
+    update.map({
+      case (id, patch) => id match {
+        case "singleton" => Right(patch)
+        case _ => Left(new IllegalArgumentException("id must be singleton"))
+      }
+    }).head
+  }
+}
+
+case class VacationResponseId(value: String)
+
+case class IsEnabled(value: Boolean) extends AnyVal
+
+case class FromDate(value: ZonedDateTime)
+
+case class ToDate(value: ZonedDateTime)
+
+case class VacationResponsePatchObject(id: VacationResponseId,
+                                       isEnabled: IsEnabled,
+                                       fromDate: Option[FromDate],
+                                       toDate: Option[ToDate],
+                                       subject: Option[String],
+                                       textBody: Option[String],
+                                       htmlBody: Option[String]) {
+  def asVacationPatch: VacationPatch = {
+    VacationPatch.builder()
+      .isEnabled(isEnabled.value)
+      .fromDate(fromDate.getOrElse(FromDate(ZonedDateTime.now())).value)
+      .toDate(toDate.getOrElse(ToDate(ZonedDateTime.now())).value)
+      .htmlBody(htmlBody.getOrElse(""))
+      .textBody(textBody.getOrElse(""))
+      .subject(subject.getOrElse(""))
+      .build()
+  }
+}
+
+case class VacationResponseSetResponse(accountId: AccountId,
+                                       newState: State,
+                                       updated: Option[Map[String, VacationResponseUpdateResponse]],
+                                       notUpdated: Option[Map[String, VacationResponseSetError]])
+
+object VacationResponseSetError {
+  val serverFailValue: String = "serverFail"
```

Comment:
This constant is already defined in "MailboxSetError". 

I would be in favour of defining only one "SetError" class that we would be using for all Method/set classes